lib/tailwindcss/commands.rb

```diff
@@ -32,6 +32,40 @@ def watch_command(always: false, poll: false, **kwargs)
       def rails_css_compressor?
         defined?(Rails) && Rails&.application&.config&.assets&.css_compressor.present?
       end
+
+      def engines_tailwindcss_roots
+        return [] unless defined?(Rails)
+
+        Rails::Engine.subclasses.select do |engine|
+          begin
+            spec = Gem::Specification.find_by_name(engine.engine_name)
```

Comment:
Is `Gem::Specification.find_by_name` guaranteed to return spec for the version of the gem that is loaded? If there are multiple versions installed on the system, could it return the spec from an earlier or later version?

I've not seen this pattern of using an engine's dependencies to control behavior, and it feels very indirect/implicit. I'm very hesitant to use this approach.

Help me understand why we shouldn't use a railtie and ask engines to be explicit/imperative? The engine, in its railtie file, should explicitly do something like:

```ruby
ActiveSupport.on_load(:tailwindcss_rails) do
  config.tailwindcss_rails.engine_roots << "path/to/application.css"
end
```

That would greatly simplify what we do in the gem.

Reply:
> Gem::Specification.find_by_name guaranteed to return spec for the version of the gem that is loaded

Yes, because it's overridden by Bundler, example:

```ruby
rails-empty(dev)> Gem::Specification.find_by_name('rails')
=>
Gem::Specification.new do |s|
  s.name = "rails"
  s.version = Gem::Version.new("8.0.2")
  s.installed_by_version = Gem::Version.new("3.6.1")
  s.authors = ["David Heinemeier Hansson"]
```